src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -548,36 +561,44 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
         : DevicePermission.PermissionDeniedByUser;
     }
     this.logger.info(`got ${kind} device for constraints ${JSON.stringify(proposedConstraints)}`);
-    const restartVideo =
-      kind === 'video' &&
-      !fromAcquire &&
-      this.boundAudioVideoController &&
-      this.boundAudioVideoController.videoTileController.hasStartedLocalVideoTile();
+
+    const oldStream: MediaStream | null = this.activeDevices[kind]
```

Comment:
So if I understand this correctly, the main fix is to release the media stream after get the new stream from GetUserMedia correct? Before we do it before getting the new stream, so all ChooseVideoInputDevice calls will release stream A (as in your example).

Reply:
Yes, in this PR, SDK is releasing an old stream ***after*** `getUserMedia()` is resolved. Before this PR, SDK released it before.